### PR TITLE
'c++_implementation_of_RSA_CRT'

### LINKED ISSUE #2021: Implementation of RSA algorithm using Chinese remainder theorem in C++
@jainaman224 @somya-kapoor @SakshiShreya @singh-shreya6 @asha15 @MastersAbh @raghav-dalmia  I would like to work on this task. Please assign me with this issue for GSSoC.

### PR DESCRIPTION
**Pull request by codermuskanj**
Fixes #2021

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
-  Implementation of RSA algorithm using Chinese remainder theorem in C++  

#### Languages Used:
- C++

#### Files Added:
- RSA_CRT.cpp

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by codermuskanj** [comment]
@jainaman224  @KavyaSharma  @sidgorey  @uday1201. I have implemented RSA_CRT algo in c++. Kindly review it.

File: RSA_CRT/RSA_CRT.cpp [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+long long int egcd (long long int a, long long int b) { 
+     int rem;
+    
+    while (b != 0) {
+        rem = a % b;
+        a = b;
+        b = rem;
+
+    }
+    return a;
```
Comment (asha15) [inline RSA_CRT/RSA_CRT.cpp:14]: please correct the indentation

File: RSA_CRT/RSA_CRT.cpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+// modular inversion function
+long long int findInv ( long long int x, long long int y, long long int n) {
```
Comment (asha15) [inline RSA_CRT/RSA_CRT.cpp:18]: remove the space after open parenthesis

File: RSA_CRT/RSA_CRT.cpp [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+      long long int d = 2, temp;
+
+    while (d < n) {
+        temp = ((d * x) - 1) % y;
+        if (temp == 0)
+        break;
+        d ++;
+    }
+    return d;
```
Comment (asha15) [inline RSA_CRT/RSA_CRT.cpp:27]: correct the indentation

File: RSA_CRT/RSA_CRT.cpp [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    cout << "Cipher = " << c << "\n";
+    
+    //printing original message
+    for (int i = 1; i <= dP; i++) {
+        m1 = ( m1 * c) % p;
+    }
+    
+    for(int i = 1; i <= dQ; i++) {
+        m2 = (m2 * c) % q;
+    }
```
Comment (asha15) [inline RSA_CRT/RSA_CRT.cpp:72]: keep a line space after this

**Comment by codermuskanj** [comment]
@asha15 @jainaman224 @KavyaSharma I have done the changes kindly go through it and please add label to my pr.

File: RSA_CRT/RSA_CRT.cpp [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+Enter the message
+1331
+Enter two prime numbers
+131
+137
+Output :
+Cipher =  16990
+Orignal Message =  1331
+*/
+
```
Comment (asha15) [inline RSA_CRT/RSA_CRT.cpp:92]: remove this line

**Comment by codermuskanj** [comment]
@asha15 I have squashed the commits and done the required changes .Kindly provide label to my pr so that it could be scored for my Gssoc.

File: RSA_CRT/RSA_CRT.cpp [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+// gcd function
+long long int egcd (long long int a, long long int b) { 
+    int rem;
+    
+    while (b != 0) {
+        rem = a % b;
+        a = b;
+        b = rem;
+
```
Comment (asha15) [inline RSA_CRT/RSA_CRT.cpp:12]: remove this line

**Comment by codermuskanj** [comment]
@jainaman224 @asha15 Do label my pull request so that it can be reflected in my Gssoc when merged. I have done the required changes kindly go through it.

**Pull request merged**